client/gutenberg/extensions/slideshow/slideshow.js

```diff
@@ -63,7 +63,7 @@ class Slideshow extends Component {
 			delay !== prevProps.delay ||
 			images !== prevProps.images
 		) {
-			const realIndex = images !== prevProps.images ? 0 : this.swiperInstance.realIndex;
+			const realIndex = jumpToIndex || this.swiperInstance.realIndex;
```

Comment:
Re https://github.com/Automattic/wp-calypso/pull/31046/files#r260475895 -- Something like

```suggestion
			let realIndex = this.swiperInstance.realIndex;
			if ( ! isEqual( images, prevProps.images ) ) {
				realIndex = prevProps.images.length;
			}
```

? (Untested)

Reply:
How about just:

`const realIndex = images.length === prevProps.images.length ? this.swiperInstance.realIndex : prevProps.images.length;`

Reply:
Sure, the length is probably a tight enough criterion here :+1:

Reply:
In fact given that this is all about determining position in the slideshow, length is probably the only meaningful criterion. Whether images are replaced doesn't really concern us here. 

Great improvement! This is done in https://github.com/Automattic/wp-calypso/pull/31046/commits/012409777bdfaca3b7b2b0b31bbe85cf4dd57225